src/test/java/com/stripe/functional/LiveStripeResponseGetterTest.java

```diff
@@ -43,4 +51,44 @@ public void testInvalidJson() throws StripeException {
     assertNotNull(exception.getCause());
     assertThat(exception.getCause(), CoreMatchers.instanceOf(JsonSyntaxException.class));
   }
+
+  private Boolean originalTelemetry;
+
+  @BeforeEach
+  public void setUp() {
+    this.originalTelemetry = Stripe.enableTelemetry;
+    Stripe.enableTelemetry = true;
+  }
+
+  @AfterEach
+  public void tearDown() {
+    Stripe.enableTelemetry = originalTelemetry;
+  }
+
+  @Test
+  public void testGlobalClientTelemetryTest() throws StripeException {
+    StripeResponseGetter responseGetterSpy =
+        Mockito.spy(new LiveStripeResponseGetter(null, httpClientSpy));
+    // We need to explicitly override the ApiResource.responseGetter = networkSpy that's set
+    // in BaseStripeTest. Unfortunately, BaseStripeTest shares the response getter spy across
+    // StripeClient and the Global Client, and so the global client will erroneously report stripe
```

Comment:
Why does the sharing matter? Isn't the usage per-request?

Reply:
Ah, well spotted. This was left over from the previous approach #1698 where it did matter. (I abandoned that approach specifically because of this ugliness). In [Move and simplify global client telemetry test](https://github.com/stripe/stripe-java/pull/1721/commits/859e4d62dfa59788bad561414a656364a48fb65c) I removed the `setStripeResponseGetter` and simplified the test, and I moved it into `TelemetryTest.java` because with the new approach it makes less sense for this to live in LiveStripeResponseGetterTest.java.